### PR TITLE
Fix/remove poly wrapper

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
### Description of the Change
poly_wrapper replaced with the shared pointer in block_loader. Also stateless validation is added to the protocol.

### Benefits
removed polywrappers

### Possible Drawbacks 
no

### CODE REVIEW

File: irohad/network/impl/block_loader_impl.hpp [hunk 1]
```diff
@@ -65,6 +70,8 @@ namespace iroha {
       std::shared_ptr<ametsuchi::PeerQuery> peer_query_;
       std::shared_ptr<ametsuchi::BlockQuery> block_query_;
       std::shared_ptr<model::ModelCryptoProvider> crypto_provider_;
+      std::shared_ptr<shared_model::validation::BlockValidator>
+          stateless_validator_;
```
Comment (Solonets) [inline irohad/network/impl/block_loader_impl.hpp]: Why did you add block stateless validation in this PR? how is it relevant to wrapper?

File: irohad/main/impl/block_loader_init.cpp [hunk 2]
```diff
@@ -30,7 +31,10 @@ auto BlockLoaderInit::createLoader(
     std::shared_ptr<BlockQuery> storage,
     std::shared_ptr<model::ModelCryptoProvider> crypto_provider) {
   return std::make_shared<BlockLoaderImpl>(
-      peer_query, storage, crypto_provider);
+      peer_query,
+      storage,
+      crypto_provider,
+      std::make_shared<shared_model::validation::BlockValidator>());
```
Comment (Solonets) [inline irohad/main/impl/block_loader_init.cpp]: this is your default value

File: irohad/network/impl/block_loader_impl.hpp [hunk 3]
```diff
@@ -35,12 +36,16 @@ namespace iroha {
       BlockLoaderImpl(
           std::shared_ptr<ametsuchi::PeerQuery> peer_query,
           std::shared_ptr<ametsuchi::BlockQuery> block_query,
-          std::shared_ptr<model::ModelCryptoProvider> crypto_provider);
+          std::shared_ptr<model::ModelCryptoProvider> crypto_provider,
+          std::shared_ptr<shared_model::validation::BlockValidator> =
```
Comment (Solonets) [inline irohad/network/impl/block_loader_impl.hpp]: BlockValidator is designed to be a template, but now it's a stub, so you should not instantiate it, as later it can change, use DefaultBlockValidator(=BlockValidator). Or should we use a template here?

File: irohad/network/impl/block_loader_impl.hpp (hunk 1 above)
Comment (Alexey-N-Chernyshov) [reply irohad/network/impl/block_loader_impl.hpp]:
```
The task is to rework poly_wrapper with the shared pointer in block_loader interface. Also, need to remove `libs/common/wrapper.hpp` because the file is redundant.

Also, has to add stateless validation of blocks in the transport layer.
```

File: irohad/main/impl/block_loader_init.cpp (hunk 2 above)
Comment (Alexey-N-Chernyshov) [reply irohad/main/impl/block_loader_init.cpp]: Yes, I want to do it explicitly because it can be changed in IR-837.
